src/CMakeLists.txt

```diff
@@ -296,6 +296,20 @@ add_subdirectory(json_spirit)
 include_directories(SYSTEM "${CMAKE_SOURCE_DIR}/src/xxHash")
 include_directories(SYSTEM "${CMAKE_SOURCE_DIR}/src/rapidjson/include")
 
+if(WITH_RADOSGW AND WITH_RADOSGW_AWS_ENDPOINT)
+  set(BUILD_ONLY "lambda" "sns")
+  set(BUILD_DEPS ON)
+  set(AWS_DEPS_INSTALL_DIR "${CMAKE_CURRENT_SOURCE_DIR}/aws-sdk-cpp/.deps" )
+  set(AUTORUN_UNIT_TESTS OFF)
+
+  add_subdirectory(aws-sdk-cpp)
+
+  include_directories(SYSTEM "${CMAKE_SOURCE_DIR}/src/aws-sdk-cpp/aws-cpp-sdk-lambda/include")
```

Comment:
please avoid adding global `include_directories()` if these include is only for certain targets.